sql: initialize virtual table descriptors concurrently; skip some vtable validations

rafiss opened the pull request:
See individual commits.

```
$ benchdiff --old 4348689e5d8 --new 959ef1f5c66 ./pkg/server -r BenchmarkTestServerStartup -c 10 -b -d 5s
old:  4348689 Merge #142011
new:  959ef1f sql: don't validate virtual tables upon creation u
args: benchdiff "--old" "4348689e5d8" "--new" "959ef1f5c66" "./pkg/server" "-r" "BenchmarkTestServerStartup" "-c" "10" "-b" "-d" "5s"

name                                   old time/op    new time/op    delta
TestServerStartup/SystemTenantOnly-24     388ms ± 4%     353ms ± 7%  -9.02%  (p=0.000 n=10+10)
TestServerStartup/ExternalTenant-24       389ms ± 4%     360ms ± 6%  -7.42%  (p=0.000 n=10+10)
TestServerStartup/SharedTenant-24         385ms ± 5%     363ms ± 6%  -5.75%  (p=0.000 n=9+10)

name                                   old alloc/op   new alloc/op   delta
TestServerStartup/SharedTenant-24         132MB ± 0%     132MB ± 0%  +0.21%  (p=0.003 n=10+10)
TestServerStartup/ExternalTenant-24       132MB ± 0%     132MB ± 0%  +0.28%  (p=0.001 n=10+10)
TestServerStartup/SystemTenantOnly-24     132MB ± 0%     132MB ± 0%  +0.39%  (p=0.000 n=9+9)

name                                   old allocs/op  new allocs/op  delta
TestServerStartup/SharedTenant-24          797k ± 0%      798k ± 0%  +0.16%  (p=0.000 n=10+10)
TestServerStartup/ExternalTenant-24        797k ± 0%      798k ± 0%  +0.19%  (p=0.000 n=10+10)
TestServerStartup/SystemTenantOnly-24      797k ± 0%      798k ± 0%  +0.21%  (p=0.000 n=10+10)
```

Epic: None

rafiss committed:
sql: remove unused field in VirtualSchemaHolder

Release note: None

rafiss committed:
sql: initialize virtual table descriptors concurrently

This helps improve TestServer startup time.

While working on this, I experimented with adding a concurrency limit,
which didn't turn out to matter. However, I still kept the addition of
SetLimit to the ctxgroup API, since it is generally useful.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/142012)
<!-- Reviewable:end -->

rafiss committed:
sql: don't validate virtual tables upon creation under tests

This patch skips an expensive validation step when running in a unit
test. The virtual tables are all initialized on startup, and validating
all of them at once leads to many heap allocations that cause TestServer
startup to take longer.

Release note: None

fqazi reviewed (approved):
Nice! 
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, 2 of 2 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/142012#-:-OK3Wbqp-kSIcH7ZeOfP:bwuv9up)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @rafiss)



<!-- Sent from Reviewable.io -->

rafiss commented:
tftr!


bors r+
